controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -175,22 +176,53 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 func (r *DataScienceClusterReconciler) reconcileSubComponent(instance *dsc.DataScienceCluster, componentName string, enabled bool,
 	component components.ComponentInterface, ctx context.Context) (*dsc.DataScienceCluster, ctrl.Result, error) {
-	err := component.ReconcileComponent(instance, r.Client, r.Scheme, enabled, r.ApplicationsNamespace)
+
+	// First set contidions to reflect a component is about to be reconciled
+	instance, err := r.updateStatus(instance, func(saved *dsc.DataScienceCluster) {
+		if enabled {
+			status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileInit, "Component reconciliation started", corev1.ConditionUnknown)
+		} else {
+			status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileInit, "Component removal started", corev1.ConditionUnknown)
+		}
+	})
 	if err != nil {
+		instance = r.reportError(err, instance, "failed to update DataScienceCluster conditions before reconciling "+componentName)
```

Comment:
here will actually call` status.SetErrorCondition(&saved.Status.Conditions, status.ReconcileFailed, fmt.Sprintf("%s : %v", message, err))`

Reply:
I am not sure I understand your comment. It this a problem? if it fails to update the status, it will try to continue as it is not a critical error.

Reply:
fixed below, as discussed.